File: Countries/one_vowel/one_vowel.py
# toy program to make a list of all the countries' names who only contain one vowel

# importing the country list
from xml.etree.ElementInclude import include


countries = open("/home/evergreen/UnderDogsAssignments/Countries/countries.txt")

#turning countries.doc into an array
countries_list = []
for line in countries:
    countries_list.append(line.strip())


#iterate through country, incriment variables for vowels, add to one_vowel as appropriate
one_vowel = []
for country in countries_list:
    lower_case = country.lower()
    if "a" in lower_case and "e" not in lower_case and "i" not in lower_case and "o" not in lower_case and not "u" in lower_case and "y" not in lower_case:
        one_vowel.append(country)
    if "e" in lower_case and "a" not in lower_case and "i" not in lower_case and "o" not in lower_case and not "u" in lower_case and "y" not in lower_case:
        one_vowel.append(country)
    if "i" in lower_case and "e" not in lower_case and "a" not in lower_case and "o" not in lower_case and not "u" in lower_case and "y" not in lower_case:
        one_vowel.append(country)
    if "o" in lower_case and "e" not in lower_case and "i" not in lower_case and "a" not in lower_case and not "u" in lower_case and "y" not in lower_case:
        one_vowel.append(country)
    if "u" in lower_case and "e" not in lower_case and "i" not in lower_case and "o" not in lower_case and not "a" in lower_case and "y" not in lower_case:
        one_vowel.append(country)

print("All the countries who's name contains exclusively one vowel are " + str(one_vowel))

File: Countries/one_vowel/oneVowel.js
// Program to create a list of all countries containing one vowel

// Importing the fs module
let fs = require("fs")

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let countries = readFileLines('/home/evergreen/UnderDogsAssignments/Countries/countries.txt');

const vowels = "AEIOUY"
var oneVowel = []

// Ex: Chad, Argentina, AU, AA

for (let j = 0; j < countries.length; j++) {
   let vowelSet = new Set ();
   let country = countries[j].toUpperCase();
   for (let k = 0; k < country.length; k++) {
      if (vowels.includes(country[k])) {
         vowelSet.add(country[k]);
      }
   }
   if (vowelSet.size === 1) {
      oneVowel.push(countries[j]);
   }
}
      // first vowel, gets added to the set
      // second vowel, is not the same, rejects and breaks to the next word
      // second vowel, is the same, proceeds to check the rest of the letters for vowels


console.log(oneVowel)